services/src/projects/project.rs

```diff
@@ -253,10 +264,57 @@ pub struct UpdateProject {
     pub id: ProjectId,
     pub name: Option<String>,
     pub description: Option<String>,
-    pub layers: Option<Vec<Option<Layer>>>,
+    pub layers: Option<Vec<LayerUpdate>>,
     pub bounds: Option<STRectangle>,
 }
 
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+#[serde(untagged)]
+pub enum LayerUpdate {
+    #[serde(deserialize_with = "deserialize_layer_update_keep")]
```

Comment:
why not just use native serialization? `-` and `/` are not intuitive either.

Reply:
What is native to you? You can make a suggestion.

Reply:
I mean the default serde serialization. "None", "Delete" I guess

Reply:
So you prefer
```
{
   "None": {}
}
```
```
{
   "Delete": {}
}
```
```
{
   "Update": {
        …
   }
}
```
?

This doesn't looks like a good API design, too, can't we do better?

Reply:
I thought
```
{
  "name": null,
  ...
  "layers": [ "None", "Delete", { "Update": { ...}]
}
```
isn't the default json representation of an payload-less enum value just a string with the variant-name?

Reply:
unfortunately not, but I can make it this way

Reply:
https://github.com/geo-engine/geoengine/pull/115/commits/ec3fb0f6bac6bbf3e76339fc1ee4d2bf0c6f7c13#diff-81d6c97ee571fac79cd18261a47f66164451154aa8f637d1b60cbc9dacea210aR280